driver-service/src/main/java/cab/aggregator/app/driverservice/config/KeycloakJwtTokenConverter.java

```diff
@@ -0,0 +1,45 @@
+package cab.aggregator.app.driverservice.config;
+
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.jwt.Jwt;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static cab.aggregator.app.driverservice.utility.Constants.AZP_CLAIM;
+import static cab.aggregator.app.driverservice.utility.Constants.AZP_CLAIM_VALUE;
+import static cab.aggregator.app.driverservice.utility.Constants.REALM_ACCESS_CLAIM;
+import static cab.aggregator.app.driverservice.utility.Constants.REALM_ACCESS_CLAIM_VALUE;
+import static cab.aggregator.app.driverservice.utility.Constants.ROLE_ADMIN;
+import static cab.aggregator.app.driverservice.utility.Constants.ROLE_PREFIX;
+
+public class KeycloakJwtTokenConverter implements Converter<Jwt, Collection<GrantedAuthority>> {
+
+    @Override
+    public Collection<GrantedAuthority> convert(Jwt jwt) {
+        if(jwt.getClaim(AZP_CLAIM).equals(AZP_CLAIM_VALUE)) {
+            return Collections.singletonList(new SimpleGrantedAuthority(ROLE_ADMIN));
+        }
+        var realmAccessMap = jwt.getClaimAsMap(REALM_ACCESS_CLAIM);
+        Object rolesObject = realmAccessMap.get(REALM_ACCESS_CLAIM_VALUE);
+
+        if (rolesObject instanceof List<?>) {
+            List<String> realmAccess = ((List<?>) rolesObject).stream()
+                    .filter(String.class::isInstance)
+                    .map(String.class::cast)
+                    .toList();
```

Comment:
JEP 394: Pattern Matching for instanceof
```suggestion

        if (rolesObject instanceof List<?> tmpRoles) {
            List<String> realmAccess = tmpRoles.stream()
                    .filter(String.class::isInstance)
                    .map(String.class::cast)
                    .toList();
```